### ISSUE TITLE
Refactor Grow and Harvest functions to reduce code duplication and improve type handling

### ISSUE THREAD

**Issue by linuxy** [linuxy/coyote-ecs#38]
The current implementation of the Grow and Harvest functions contains significant code duplication with nearly identical conditional blocks for each fruit type. This makes the code harder to maintain and introduces potential bugs when adding new fruit types.

Issues:
1. The Grow function doesn't handle errors properly despite using `try` (this will cause runtime failures)
2. Both functions use multiple if-statements to check component types instead of a more generic approach
3. Code duplication across fruit types in both functions
4. In the Grow function, components are detached unconditionally after processing

Proposed solution:
1. Create a generic function to handle components of different fruit types
2. Use a compile-time approach to iterate over fruit component types
3. Properly handle errors in the Grow function
4. Add a comment explaining the detach logic (or make it conditional if needed)

Example refactored Grow function:
```zig
pub fn Grow(world: *World) !void {
    var it = world.components.iterator();
    var i: u32 = 0;
    
    while (it.next()) |component| : (i += 1) {
        inline for (@typeInfo(Components).Struct.decls) |decl| {
            const FruitType = @field(Components, decl.name);
            if (component.is(FruitType)) {
                try component.set(FruitType, .{ .ripe = true });
                break;
            }
        }
        // Fruits fall from the tree
        component.detach();
    }
    
    std.log.info("Fruits grown: {}", .{i});
}
```

The Harvest function would be refactored similarly to reduce duplication and improve maintainability, especially when adding new fruit types in the future.
